hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -207,6 +208,11 @@ public KafkaOffsetGen(TypedProperties props) {
     maxEventsToReadFromKafka = (maxEventsToReadFromKafka == Long.MAX_VALUE || maxEventsToReadFromKafka == Integer.MAX_VALUE)
         ? Config.maxEventsFromKafkaSource : maxEventsToReadFromKafka;
     long numEvents = sourceLimit == Long.MAX_VALUE ? maxEventsToReadFromKafka : sourceLimit;
+
+    if (numEvents < toOffsets.size()) {
```

Comment:
Do you mean something like setting sourceLimit to 5 but you have 10 kafka partitions? Why setting sourceLimit to this small?

Reply:
The general production environment will not have such operations. But I found this problem in the test environment, and feel that I should give at least one hint

Reply:
Delta streamer has `--continues` mode, which means it is intended to not consuming data in some cases. Throwing an exception here will interrupt the application unexpectedly. 
If the goal is to provide a hint to the user, I think we can add a log to `computeOffsetRanges`.

Reply:
Hi @garyli1019, IMO, throwing an exception makes sense, because the 'sourceLimit' is one-time config binding with the job. If the 'sourceLimit' is lesser than the partitions of Kafka, the job will never consume data from Kafka, running correctly is meaningless.

Reply:
I think if people really set `sourceLimit` to 0 then they should consume no new data, which is intended behavior, instead of throwing an exception. If they want to consume data then they should set this number higher.

Reply:
@pratyakshsharma thoughts?

Reply:
Interesting! I guess both of you have valid points but am somewhat more inclined towards actually throwing the exception. The reason being throwing this exception will help users detect misconfigs around sourceLimit, for example, if someone sets sourceLimit to very small number in production setup by mistake :).
Also would like to hear from @vinothchandar on this :)

Reply:
@garyli1019 maybe adding a warning log to `computeOffsetRanges` and giving some reminders on the description of `--source-limit` is also doable, WDYT?

Reply:
Thanks for all the feedback here! I don't have a strong opinion here and we have more votes on throwing the exception, so let's go for it. I agree this issue is most likely caused by misconfigs.